CMakeLists.txt

```diff
@@ -1,9 +1,16 @@
 cmake_minimum_required(VERSION 3.20.0)
 
 set(APP "zephyr_app_scratch")
-
+set(CXX_STANDARD C++17)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 
 project(${APP})
 
-target_sources(app PRIVATE src/main.c)
\ No newline at end of file
+target_sources(app PRIVATE src/main.cpp)
+
+set(ALLOWED_BOARDS "nrf5340dk_nrf5340_cpuapp;native_posix_64")
+list(FIND ALLOWED_BOARDS ${BOARD} BOARD_INDEX)
+if(BOARD_INDEX EQUAL -1)
+  message(FATAL_ERROR 
+    "This application is intended to be built only for the following boards: ${ALLOWED_BOARDS}. Current BOARD is: ${BOARD}")
```

Comment:
Very nice! But I simply assumed comparing CONFIG_BAORD definition in the source file (main.cpp) with the one of these strings. My fault :(

Reply:
thanks! will review in the following PR